ggshield/core/config/user_config.py

```diff
@@ -63,14 +64,81 @@ def validate_policy_id(policy_id: str) -> bool:
     return bool(POLICY_ID_PATTERN.fullmatch(policy_id))
 
 
-def validate_policy_ids(values: Iterable[str]) -> None:
-    invalid_excluded_policies = [
-        policy_id for policy_id in values if not validate_policy_id(policy_id)
-    ]
-    if len(invalid_excluded_policies) > 0:
-        raise ValidationError(
-            f"The policies {invalid_excluded_policies} do not match the pattern '{POLICY_ID_PATTERN.pattern}'"
-        )
+@marshmallow_dataclass.dataclass
+class ConfigIgnoredElement(FilteredConfig):
+    comment: Optional[str]
+    until: Optional[datetime]
+
+    @abstractmethod
+    def short_repr(self):
+        return "Element"
+
+    # Accept date yyyy-mm-dd instead of a full datetime
+    @pre_load
+    def parse_date(self, data, **kwargs):
+        if not (isinstance(data, dict)) or data.get("until") is None:
+            return data
+        try:
+            data["until"] = str(datetime.strptime(data["until"], "%Y-%m-%d"))
+            return data
+        except ValueError:
+            return data
+
+    @post_load
+    def datetime_to_utc(self, data, **kwargs):
+        if data["until"] is not None:
+            data["until"] = data["until"].astimezone(timezone.utc)
+        return data
+
+
+def validate_ignored_elements_array(data: dict, field: str):
```

Comment:
This function does too much: it alter `data` in place, show warnings and return the modified `data`. I'd rather split it in two, a function to take expired elements from the list and return them, and another to report the expired elements. Something like this (untested):

```python
def remove_expired_elements(lst: List[ConfigIgnoredElement]) -> List[ConfigIgnoredElement]:
    expired: List[ConfigIgnoredElement] = []
    now = datetime.now(tz=timezone.utc)
    for idx in range(len(lst) - 1, -1, -1):
        ignored = lst[idx]
        if ignored.until and ignored.until <= now:
            lst.pop(idx)
            expired.insert(0, ignored)
    
    return expired
    

def report_expired_elements(expired_lst: List[ConfigIgnoredElement]) -> None:
    for element in expired_lst:
        display_warning(
            f"{element.short_repr()} has an expired 'until' "
            f"date ({element.until}), please update your configuration file."
        )
```

Then it can be used like that:

```
    def validate_ignored_vulns(self, data, **kwargs):
        expired_lst = remove_expired_elements(data["ignored_vulnerabilities"])
        report_expired_elements(expired_list)
        return data["ignored_vulnerabilities"]
```